### PR TITLE
Add checks for bookmarks & fix bad links

### PR DESCRIPTION
**Pull request by duglin**
- Add support for href checking bookmarks & fixed bad links
- Speed up the checkers.
- Change Makefile so that it checks ALL files w/o needed to state which ones. So, now we have a flag in each md file that we DON'T want checked instead. We were missing some files. Look for the `<!-- no verify-specs -->`  line
- fixed some RFC keywords that were not checked before

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

File: cloudevents/bindings/amqp-protocol-binding.md [hunk 1]
```diff
@@ -327,3 +327,8 @@ content-type: application/cloudevents+json; charset=utf-8
 [message-format]: http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#section-message-format
 [data]: http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-data
 [app-properties]: http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-application-properties
+[amqp-boolean]: http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-boolean
+[amqp-long]: http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-long
+[amqp-binary]: http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-binary
+[amqp-string]: http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-string
+[amqp-timestamp]: http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-timestamp
```
Comment (duglin) [inline cloudevents/bindings/amqp-protocol-binding.md:334]: @clemensv can you check these links?

File: cloudevents/bindings/mqtt-protocol-binding.md [hunk 2]
```diff
@@ -319,12 +319,14 @@ Topic Name: mytopic
...
-[ce-types]: ./spec.md#type-system
-[json-format]: ./formats/json-format.md
+[ce]: ../spec.md
+[ce-types]: ../spec.md#type-system
+[json-format]: ../formats/json-format.md
 [oasis-mqtt-3.1.1]: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/mqtt-v3.1.1.html
 [oasis-mqtt-5]: http://docs.oasis-open.org/mqtt/mqtt/v5.0/mqtt-v5.0.html
+[3-publish]: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/errata01/os/mqtt-v3.1.1-errata01-os-complete.html#_Toc442180850
 [5-content-type]: http://docs.oasis-open.org/mqtt/mqtt/v5.0/mqtt-v5.0.html#_Toc502667341
+[5-publish]: https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901100
```
Comment (duglin) [inline cloudevents/bindings/mqtt-protocol-binding.md:329]: @clemensv can you check these links?

File: cloudevents/bindings/websockets-protocol-binding.md [hunk 3]
```diff
@@ -154,15 +154,17 @@ format specification and the resulting data becomes the payload.
...
+[json-batch-format]: ../formats/json-format.md#4-json-batch-format
+[avro-format]: ../formats/avro-format.md
+[proto-format]: ../formats/protobuf-format.md
 [rfc2119]: https://tools.ietf.org/html/rfc2119
 [rfc6455]: https://tools.ietf.org/html/rfc6455
 [rfc6455-section-1-3]: https://tools.ietf.org/html/rfc6455#section-1.3
 [rfc6455-section-4]: https://tools.ietf.org/html/rfc6455#section-4
 [rfc6455-section-1-9]: https://tools.ietf.org/html/rfc6455#section-1.9
+[rfc7230-section-5-1]: https://datatracker.ietf.org/doc/html/rfc7230#section-5.1
+[rfc6455-section-5-1]: https://datatracker.ietf.org/doc/html/rfc6455#section-5.1
```
Comment (duglin) [inline cloudevents/bindings/websockets-protocol-binding.md:170]: Can someone double-check these?

File: misc/Makefile [hunk 4]
```diff
@@ -5,22 +5,6 @@ verify:
...
-		cloudevents/bindings/mqtt-protocol-binding.md \
-		cloudevents/bindings/nats-protocol-binding.md \
-		cloudevents/formats/protobuf-format.md \
-		cloudevents/spec.md \
-		cloudevents/bindings/websockets-protocol-binding.md \
-		\
-		discovery/spec.md \
-		pagination/spec.md \
-		subscriptions/spec.md 
+	@misc/tools/verify-specs.sh -v .
```
Comment (duglin) [inline misc/Makefile:8]: check all files by default now

File: schemaregistry/spec.md [hunk 5]
```diff
@@ -26,7 +26,7 @@ deserialize the event or message payload need access to the schema used to
 publish it.
 
 For formats that do not depend on external schemas for serialization, schemas
-might still be required in some scenarios to allow for consumers or inspecting
+might still be needed in some scenarios to allow for consumers or inspecting
```
Comment (duglin) [inline schemaregistry/spec.md:29]: @clemensv can you check the RFC keywords in this file?

**Comment by duglin** [comment]
ping @clemensv - to check the links

File: cloudevents/bindings/amqp-protocol-binding.md (hunk 1 above)
Comment (clemensv) [reply cloudevents/bindings/amqp-protocol-binding.md:334]: LGTM

File: cloudevents/bindings/websockets-protocol-binding.md (hunk 3 above)
Comment (clemensv) [reply cloudevents/bindings/websockets-protocol-binding.md:170]: LGTM

File: schemaregistry/spec.md (hunk 5 above)
Comment (clemensv) [reply schemaregistry/spec.md:29]: If you need a review for those let's make that an issue.

**Comment by clemensv** [comment]
I should have caught them all

**Comment by duglin** [comment]
Approved on the 1/13 call pending Clemens check of the links, which we reviewed/approved

**Pull request merged**